jgrapht-core/src/main/java/org/jgrapht/graph/AbstractBaseGraph.java

```diff
@@ -293,6 +288,20 @@ public boolean addEdge(V sourceVertex, V targetVertex, E e)
         }
     }
 
+    private boolean addIntrusiveEdge(E e, V sourceVertex, V targetVertex)
```

Comment:
I prefer to keep two methods with similar code than refactoring to a function here. The reason is that this method reverts changes that happen else where.

Reply:
My intention was not only to avoid code duplication but also to increase readability/comprehensibility of the code.
But I also understand your argument, so I changed it accordingly.